ui/src/message_popup/mobile_prototype/mobile_interface.jsx

```diff
@@ -0,0 +1,375 @@
+      helpType: React.PropTypes.string.isRequired,
+      shouldShowStudentCard: React.PropTypes.bool.isRequired,
+      shouldShowSummary: React.PropTypes.bool.isRequired,
+    }).isRequired,
+    question: React.PropTypes.object.isRequired,
+    onQuestionDone: React.PropTypes.func.isRequired,
+    limitMs: React.PropTypes.number.isRequired,
+    onLog: React.PropTypes.func.isRequired,
+    isLastQuestion: React.PropTypes.bool.isRequired,
+  },
+  
+  addMessages(messageList){
+    var messages = _.clone(this.state.messages);
+    var messageId = this.state.messageId;
+    for(var messageIndex = 0; messageIndex < messageList.length; messageIndex++){
+      messageId+=1;
+      const message = _.extend({key: messageId}, messageList[messageIndex]);
+      messages.push(message);
+    }
+    this.setState({messages, messageId});
+  },
+  
+  addResponse(response){
+    this.addMessages([{type: 'user', text: response}]);
+  },
+
+  setInitialResponse(response){
+    this.setState({initialResponse: response});
+    if(this.props.scaffolding.helpType === 'feedback'){
+      this.sendRevision(response);
+    }else{
+      this.addResponse(response);
+    }
+  },
+
+  setRevisedResponse(response){
+    this.addResponse(response);
+    this.setState({revisedResponse: response});
+  },
+
+  setPassedResponse(){
+    this.setState({revisedResponse: ''});
+  },
+  
+  isReadyToMoveOn(){
+    if(this.props.scaffolding.helpType === 'feedback'){
+      if(this.state.revisedResponse !== undefined){ 
+        return true;
+      }
+    }else{
+      if(this.state.initialResponse !== undefined){
+        return true;
+      }
+    }
+    return false;
+  },
+  
+  onNextQuestion(){
+    this.props.onQuestionDone(this.state.elapsedMs);
+    this.setState(this.getInitialState());
+  },
+  
+  sendRevision(response){
+    if(this.props.question.examples.length > 0){
+      this.addMessages([
+        {type: 'user', text: response},
+        {type: 'info', text: "Here's an example of something you could say/do:\n\n" + _.shuffle(this.props.question.examples)[0] + "\n\nWould you like to revise your response?"}, 
+      ]);
+    }else{
+      this.addMessages([
+        {type: 'user', text: response},
+        {type: 'info', text: "There aren't any examples for this situation yet but your response will be recorded for feedback."}, 
+      ]);
+      this.setPassedResponse();
+    }
+  },
+  
+  showExample(){
+    const example = this.nextExample(true);
+    const messageText = example.type + ' Example:\n' + example.text;
+    this.addMessages([{type: 'info', text: messageText}]);
+  },
+  
+  nextExample(increment){
+    const currentIndex = this.state.exampleIndex;
+    if(this.state.allExamples.length > currentIndex){
+      if(increment){
+        this.setState({exampleIndex: this.state.exampleIndex + 1});
+      }
+      return this.state.allExamples[currentIndex];
+    }else{
+      return null;
+    }
+  },
+
+  setDialog(value){
```

Comment:
I'd remove this, I'm not sure it's adding much to say `setDialog(value)` over `setState({dialog: value})`.